x-pack/legacy/plugins/ml/public/jobs/new_job_new/pages/components/pick_fields_step/components/advanced_detector_modal/advanced_detector_modal.tsx

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, Fragment, useState, useContext, useEffect } from 'react';
+import {
+  EuiComboBox,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiFlexGrid,
+  EuiComboBoxOptionProps,
+  EuiHorizontalRule,
+} from '@elastic/eui';
+import { JobCreatorContext } from '../../../job_creator_context';
+import {
+  AdvancedJobCreator,
+  isAdvancedJobCreator,
+  JobCreatorType,
+} from '../../../../../common/job_creator';
+import {
+  Field,
+  Aggregation,
+  EVENT_RATE_FIELD_ID,
+} from '../../../../../../../../common/types/fields';
+import { RichDetector } from '../../../../../common/job_creator/advanced_job_creator';
+import { ES_FIELD_TYPES } from '../../../../../../../../../../../../src/plugins/data/public';
+import { ModalWrapper } from './modal_wrapper';
+import { ES_AGGREGATION } from '../../../../../../../../common/constants/aggregation_types';
+import {
+  AggDescription,
+  FieldDescription,
+  ByFieldDescription,
+  OverFieldDescription,
+  PartitionFieldDescription,
+  ExcludeFrequentDescription,
+} from './descriptions';
+
+interface Props {
+  payload: ModalPayload;
+  fields: Field[];
+  aggs: Aggregation[];
+  detectorChangeHandler: (dtr: RichDetector, index?: number) => void;
+  closeModal(): void;
+}
+
+export interface ModalPayload {
+  detector: RichDetector;
+  index?: number;
+}
+
+const emptyOption: EuiComboBoxOptionProps = {
+  label: '',
+};
+
+const mlCategory: Field = {
+  id: 'mlcategory',
+  name: 'mlcategory',
+  type: ES_FIELD_TYPES.KEYWORD,
+  aggregatable: false,
+};
+
+const excludeFrequentOptions: EuiComboBoxOptionProps[] = [{ label: 'all' }, { label: 'none' }];
+
+export const AdvancedDetectorModal: FC<Props> = ({
```

Comment:
Should any script_fields set in the datafeed step be available in the detector modal? I configured `script_fields` but the field I added didn't seem to be available when configuring a detector.

Reply:
yep!
https://github.com/elastic/kibana/projects/30#card-27612562